coops-ui/tests/unit/Dashboard.spec.ts

```diff
@@ -142,19 +149,47 @@ describe('Dashboard - Click Tests', () => {
   it('routes to Standalone Office Address Filing page when EDIT is clicked', done => {
     // init store
     store.state.entityIncNo = 'CP0001191'
+    // create a Local Vue and install router on it
+    const localVue = createLocalVue()
+    localVue.use(VueRouter)
+    const router = mockRouter.mock()
+    const wrapper = shallowMount(Dashboard, { localVue, store, router, vuetify })
+    const vm = wrapper.vm as any
+
+    Vue.nextTick(async () => {
+      const button = vm.$el.querySelector('#btn-standalone-addresses')
+      expect(button.textContent).toContain('Change')
+      await button.click()
 
+      // verify routing to Standalone Office Address Filing page with id=0
+      expect(vm.$route.name).toBe('standalone-addresses')
+      expect(vm.$route.params.id).toBe(0)
+
+      wrapper.destroy()
+      done()
+    })
+  })
+
+  it('displays the change of address warning dialog as a bcorp', done => {
+    // init store
+    store.state.entityIncNo = 'BC0007291'
+    store.state.entityType = EntityTypes.BCorp
     // create a Local Vue and install router on it
     const localVue = createLocalVue()
     localVue.use(VueRouter)
     const router = mockRouter.mock()
-    const wrapper = mount(Dashboard, { localVue, store, router, vuetify })
+    const wrapper = shallowMount(Dashboard, { localVue, store, router, vuetify })
     const vm = wrapper.vm as any
+    vm.coaWarning = false
 
     Vue.nextTick(async () => {
       const button = vm.$el.querySelector('#btn-standalone-addresses')
-      expect(button.querySelector('.v-btn__content').textContent).toContain('Change')
+      expect(button.textContent).toContain('Change')
       await button.click()
 
+      expect(vm.coaWarning).toBe(true)
```

Comment:
This doesn't prove the dialog is displaying. Is it possible to check that the dialog element(s) are defined and not display=none (or something like that)?

I agree that it's probably not worth having a separate unit test component for the dialog component, but if you think it's worthwhile then you could see how SaveErrorDialog.spec.ts does it.

Reply:
Good point, i will update this!

Reply:
How is this? I used the stubbed child component to emit the value to proceed with the routing?

Reply:
After trying a few things myself, I find this is a decent way to do it.

As you know, the problem is that dialogs are injected dynamically at the top-most level of the Vue app element tree, and so you cannot "see" them from within a component further down the tree (as when testing that component in a unit test).

So, your approach to exercise the dialog and check for results seems OK. Thanks for this.

Reply:
Next time please add comments explaining what your code is doing... especially in this case since it appears to be doing something unusual.